devicetypes/smartthings/zwave-dimmer-switch-generic.src/zwave-dimmer-switch-generic.groovy

```diff
@@ -185,6 +192,13 @@ def poll() {
 	zwave.switchMultilevelV1.switchMultilevelGet().format()
 }
 
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	refresh()
```

Comment:
It tries to fetch the state variable "MSR" if it exists. 
If it doesn't exist, then inside the if section, MSR(Manufacturer Specific Report) is fetched from the device by sending the corresponding z-wave command. When this takes place a new state variable "MSR" is created containing the appropriate manufacturer specific details obtained from the device. So if getDataValue("MSR") is called later on, this value can be directly read from the state variable "MSR".

Overall, this doesn't affect the DTH from generating an event.